cli/codegen.md

```diff
@@ -0,0 +1,46 @@
+<a href="https://aws-amplify.github.io/" target="_blank">
+    <img src="https://s3.amazonaws.com/aws-mobile-hub-images/aws-amplify-logo.png" alt="AWS Amplify" width="550" >
+</a>
+
+# AWS Amplify CLI codegen configuration & Workflow
+
+Codegen helps you generate native code for iOS and android, generation of types for Flow and Typescript. It can also generates GraphQL statements(queries, mutations and subscriptions) so that you doesn't have to hand code them.
+
+Codegen `add` workflow triggers automatically when an AppSync API is pushed to the cloud. You will be prompted if you want to configure codegen when an AppSync API is created and if you opt-in for codegen, subsequent pushes prompts you if they want to update the generated code after changes get pushed to cloud.
+
+When an project is configured to generate code with codegen, it stores all the configuration `.graphqlconfig.yml` file in the root folder of your project. Codegen behind the scene use [`amplify-graphql-docs-generator`](https://www.npmjs.com/package/amplify-graphql-docs-generator) to generate the GraphQL statements and [`aws-appsync-codegen`](https://www.npmjs.com/package/aws-appsync-codegen) to generate the types.
```

Comment:
@yuth can you chat with @mlabieniec on the aws-appsync-codegen reference here? I get that it works like this now but really we want to pull this package in (or port over the code) for maintainability so maybe not point to this otherwise customers will look to that code for issues?

Reply:
The thing is, we've already started with a banner there. Let's talk timeline with Mike on the other stuff.